### ISSUE TITLE
[Bug]Fix retain cycle issue

### ISSUE THREAD

**Issue by istroot** [exyte/PopupView#42]
First, thanks for your PopupView library❤️.

## Issue
ContentView's @StateObject variable never deinit

## Minimal Example
```
class Test: ObservableObject {
    var loadState: EMLoadState = .loading
    var groupEntityArr: [ListGroupEntity] = []
    var groupStructArr: [ListGroupStruct] = []
    @Published var groupUIArr: [ListGroupUI] = []
    
    deinit {
        print("deinit will never be executed") // 《=========
    }
}
struct TestView: View {
    @StateObject var viewModel = Test()
    @State var show: Bool = false
    @State var text: String = ""
    var body: some View {
        ZStack {
            
        }
        .popup(isPresented: $show, autohideIn: 2, dismissCallback: {
            temp()
        }, view: {
            Text(text)
        })
    }
    
    func temp() {
        print(self.text)
    }
}
```

## Solution
```
if let autohideIn = autohideIn {
            dispatchWorkHolder.work?.cancel()
            
            let block = dismissCallback // add this code
            dispatchWorkHolder.work = DispatchWorkItem(block: { [weak isPresentedRef] in
                isPresentedRef?.value.wrappedValue = false
                dismissCallback() // remove this code, which cause @StateObject never deinit
                block() // add this code
            })
            if isPresented, let work = dispatchWorkHolder.work {
                DispatchQueue.main.asyncAfter(deadline: .now() + autohideIn, execute: work)
            }
        }
```

**Comment by manindaniil** [comment]
Hey @istroot ! Thank you so much for reporting this issue !
We added your fix to the master.

**Comment by f3dm76** [comment]
Hey @istroot , I'm happy to announce that your fix was included in 1.0.2 release and is now live! Have a wonderful day!
